src/fromager/dependencies.py

```diff
@@ -32,10 +32,15 @@ def get_build_system_dependencies(
         )
         return _read_requirements_file(build_system_req_file)
 
-    dep_func = overrides.find_override_method(req.name, "get_build_system_dependencies")
-    if not dep_func:
-        dep_func = default_get_build_system_dependencies
-    deps = _filter_requirements(req, dep_func(ctx, req, sdist_root_dir))
+    orig_deps = overrides.find_and_invoke(
+        req.name,
+        "get_build_system_dependencies",
+        default_get_build_system_dependencies,
```

Comment:
This smells like a violation of DRY (don't repeat yourself). How about we enforce a consistent naming schema for default methods, then change `find_and_invoke` to get the name from the default function name?

```
assert default_fn.__name__.startswith("default_")
method = default_fn.__name__[8:]
```

Reply:
If we enforce this naming convention we will lose the opportunity to have common default functions or default functions that might have other use as well

Reply:
I need this feature so I can update fromager in the builder repo today. Let's consider this improvement in a follow-up PR.